persistence-modules/spring-data-jpa/src/test/java/com/baeldung/dao/repositories/PersonInsertRepositoryIntegrationTest.java

```diff
@@ -0,0 +1,103 @@
+
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+@RunWith(SpringRunner.class)
+@DataJpaTest(excludeAutoConfiguration = {PersistenceConfiguration.class, PersistenceUserConfiguration.class, PersistenceProductConfiguration.class})
+public class PersonInsertRepositoryIntegrationTest {
+
+    private static final Long ID = 1L;
+    private static final String FIRST_NAME = "firstname";
+    private static final String LAST_NAME = "firstname";
+    private static final Person PERSON = new Person(ID, FIRST_NAME, LAST_NAME);
+
+    @Autowired
+    private PersonRepository personRepository;
+    @Autowired
+    private PersonCustomRepository personCustomRepository;
+
+    @BeforeEach
+    public void clearDB() {
+        personRepository.deleteAll();
+    }
+
+    @Test
+    public void givenPersonEntity_whenSaveWithNativeQuery_ThenPersonIsPersisted() {
+        insertPerson();
+
+        assertPersonPersisted();
+    }
+
+    @Test
+    public void givenPersonEntity_whenInsertedTwiceWithNativeQuery_thenDataIntegrityViolationExceptionIsThrown() {
+        assertThatExceptionOfType(DataIntegrityViolationException.class).isThrownBy(() -> {
+            insertPerson();
+            insertPerson();
+        });
+    }
+
+    @Test
+    public void givenPersonEntity_whenSaveWithQueryAnnotation_thenPersonIsPersisted() {
+        insertPersonWithQueryAnnotation();
+
+        assertPersonPersisted();
+    }
+
+    @Test
+    public void givenPersonEntity_whenInsertedTwiceWithQueryAnnotation_thenDataIntegrityViolationExceptionIsThrown() {
+        assertThatExceptionOfType(DataIntegrityViolationException.class).isThrownBy(() -> {
+            insertPersonWithQueryAnnotation();
+            insertPersonWithQueryAnnotation();
+        });
+    }
+
+    @Test
+    public void givenPersonEntity_whenSaveWithEntityManager_thenPersonIsPersisted() {
+        insertPersonWithEntityManager();
+
+        assertPersonPersisted();
+    }
+
+    @Test
+    public void givenPersonEntity_whenInsertedTwiceWithEntityManager_thenDataIntegrityViolationExceptionIsThrown() {
+        assertThatExceptionOfType(DataIntegrityViolationException.class).isThrownBy(() -> {
+            insertPersonWithEntityManager();
+            insertPersonWithEntityManager();
+        });
+    }
+
+    private void insertPerson() {
+        personRepository.insert(PERSON);
+    }
+
+    private void insertPersonWithQueryAnnotation() {
+        personRepository.insertWithAnnotation(ID, FIRST_NAME, LAST_NAME);
+    }
+
+    private void insertPersonWithEntityManager() {
+        personCustomRepository.insert(new Person(ID, FIRST_NAME, LAST_NAME));
+    }
+
+    private void assertPersonPersisted() {
+        Optional<Person> personOptional = personRepository.findById(PERSON.getId());
+        assertThat(personOptional.isPresent()).isTrue();
```

Comment:
Per our docs `always use a new line before the then section of the test`. I know this isn't technically a test, but it would still be nice to follow that pattern.

Reply:
Fair point. I used a new line for every usage of assertPersonPersisted() method but you are right, this should be treated as assertion as well.